### PR TITLE
add required_plugins to ForemanAnsibleModule

### PR DESCRIPTION
**Pull request by evgeni**
this PR adds `required_plugins` as a parameter to `ForemanAnsibleModule`, which allows individual modules to express a dependency on Foreman plugins, either for the whole module, or for individual parameters.

the syntax is:
```python
required_plugins=[
    ('plugin_name', ['list', 'of', 'params']),
    ('plugin_name', ['*']),
]
```

Fixes: #728

### CODE REVIEW

**Comment by evgeni** [comment]
@ephracis for your specific problem, this would have generated the following output:

```
The server is missing required plugins: discovery (for discovery_proxy).
```

Would that have been helpful? :)

**Comment by evgeni** [comment]
(I'd like to hear back from Christoffer before merging this if it gets approved before that)

File: plugins/module_utils/foreman_helper.py [hunk 1]
```diff
@@ -261,7 +273,7 @@ def _patch_subnet_rex_api(self):
             see https://projects.theforeman.org/issues/19086
         """
 
-        if 'remote_execution_features' not in self.foremanapi.apidoc['docs']['resources']:
+        if not self.has_plugin('remote_execution'):
```
Comment (mdellweg) [inline plugins/module_utils/foreman_helper.py]: We could move this patching to the place where we check for the availability of the plugin, right? No need to patch, if it is not used...

File: plugins/module_utils/foreman_helper.py [hunk 2]
```diff
@@ -79,7 +88,7 @@ class KatelloMixin(OrganizationMixin):
     def connect(self):
         super(KatelloMixin, self).connect()
 
-        if 'subscriptions' not in self.foremanapi.resources:
+        if not self.has_plugin('katello'):
```
Comment (mdellweg) [inline plugins/module_utils/foreman_helper.py]: Would it be possible to just add 'katello' to the `required_plugins`?

Comment (evgeni) [reply plugins/module_utils/foreman_helper.py]: Yeah, I considered that, would add ~5 lines and remove 2 XD

File: plugins/module_utils/foreman_helper.py (hunk 1 above)
Comment (evgeni) [reply plugins/module_utils/foreman_helper.py]: You mean only patch if the param is used?

Comment (mdellweg) [reply plugins/module_utils/foreman_helper.py]: I meant only patch if the plugin is required.

**Comment by ephracis** [comment]
Yes, that message would have been plenty helpful.

File: plugins/module_utils/foreman_helper.py (hunk 1 above)
Comment (mdellweg) [reply plugins/module_utils/foreman_helper.py]: This is not going to be a blocker for me.

File: plugins/module_utils/foreman_helper.py (hunk 2 above)
Comment (evgeni) [reply plugins/module_utils/foreman_helper.py]: implemented

File: plugins/module_utils/foreman_helper.py (hunk 1 above)
Comment (evgeni) [reply plugins/module_utils/foreman_helper.py]: I'd like to leave that out for now. Feel free to open a follow up ;)

**Comment by evgeni** [comment]
now with :100: more docs!

**Pull request merged**